render/gif.go

```diff
@@ -102,54 +104,3 @@ func recoverToError(panicArg interface{}) error {
 	err = fmt.Errorf("panic at %s: %w", source, err)
 	return err
 }
-
-// getColorCounts finds all unique colours in an image and returns a count
```

Comment:
bye bye naive Torben algorithm :')

Reply:
Any idea if median cut is faster/slower?

Reply:
I haven't benchmarked it. The library is supposed to be a pretty fast implementation. I think it's doing a similar sort of thing that my algorithm did, but a little more clever and better re-use of data structures. I wouldn't be surprised if it's even faster than my algorithm was.

Reply:
Hmm, it actually may be a decent amount slower

Reply:
For a quick idea of how it compares, I wrote a bash script to export 10 different games from a cold-start exporter. Once for my old algorithm and once for the median cut algorithm. Here's what I got:

|                  Median Cut                  |                  Torb Naive                  |                 Game                 |
|----------------------------------------------|----------------------------------------------|--------------------------------------|
| duration=8.95548023s fps=37.51892599510546   | duration=4.556901875s fps=73.73430660057826  | a303d059-3b49-473c-a065-9cc8609fc987 |
| duration=10.984440832s fps=41.69534043697055 | duration=4.521394342s fps=101.2961855031224  | 8ba59ad2-25d4-4c1e-aa9f-2ebf0f83bac1 |
| duration=7.461391551s fps=36.18627948345824  | duration=3.990670464s fps=67.65780397947687  | 9a47a802-d86b-499c-b652-9bce3aa6e588 |
| duration=2.293973633s fps=27.027337676465784 | duration=1.727955s fps=35.88056401931763     | 96aacb2a-46cf-4ff2-a717-929a951c67f0 |
| duration=3.128326887s fps=13.745366629903597 | duration=2.380697879s fps=18.06193065457845  | ad71b633-cb7e-488c-bde2-d654a2411a71 |
| duration=2.492721384s fps=25.273582681312607 | duration=1.583533083s fps=39.784454569554455 | 0da85829-0c49-40b2-8938-5475610b1ef9 |
| duration=1.236328907s fps=8.897308748277936  | duration=911.810476ms fps=12.06391052695034  | 79049d0f-5dfc-4e89-bf5c-a26541695a40 |
| duration=4.060325268s fps=28.815425434531964 | duration=2.459846923s fps=47.5639353433051   | 130d3031-9e8c-4089-882b-23b3a7a45524 |
| duration=4.068599396s fps=31.460457897585552 | duration=2.595810322s fps=49.31022845358729  | 985e51bd-44e5-47af-acfd-dafd45852b28 |



**Note:** this isn't the cleanest benchmark because things can run in parallel and other stuff on my system may be adding some noise. But I think it's probably not enough noise to explain the pretty significant difference.

Reply:
It seems to have roughly halved the FPS. Longer games are a better comparison of the algorithms because the FPS is less influenced by the first frame render, which caches things and customizes assets.